lib/Worker.js

```diff
@@ -149,13 +162,18 @@ module.exports = function(redis) {
 			// remove from processing list
 			.lrem(self.master.id + ':processing', 1, task)
 
+			// remove from retry list
+			.zrem(self.master.id + ':retries', task)
+
 			.exec(function(err){
 				if(err){
 
 					// TODO: differentiate between errors in lpush and lrem
 
 					self.master.emit('error', err, 'error deadlettering task');
 					self.master.emit('error', new DeadletterError(task));
+
+					// save a reference to this timer somewhere so it can be cleared on stop?
```

Comment:
Yes, your suggestion is definitely the correct behavior here.

Reply:
Actually, wouldn't we want these redis ops to be retried for as long as they could be?

Reply:
Canceling a `setTimeout` or `setImmediate` ID won't stop an active redis operation; it will just prevent the worker from proceeding to its next phase. Once all workers are stopped, the swarm can report that it's stopped, and the process can exit without potentially interrupting a new redis operation.